packages/example/src/mocks/handlers/documents.ts

```diff
@@ -1,44 +1,40 @@
-import { rest } from 'msw';
+import { http, HttpResponse, delay } from 'msw';
 import {
   GetDocumentsResponse,
   GetDocumentsParams,
 } from '../../types/documents';
 
 export default [
-  rest.get<null, GetDocumentsParams, GetDocumentsResponse>(
+  http.get<GetDocumentsParams, GetDocumentsResponse>(
     '/api/documents/months',
-    (_, response, context) =>
-      response(
-        context.delay(50),
-        context.status(200),
-        context.json([
-          {
-            id: '9ef0aeb6-60bb-11ed-9b6a-0242ac120002',
-            title: 'january.pdf',
-          },
-          {
-            id: 'aed3c728-60bb-11ed-9b6a-0242ac120002',
-            title: 'february.pdf',
-          },
-          {
-            id: 'b60a4418-60bb-11ed-9b6a-0242ac120002 ',
-            title: 'march.pdf',
-          },
-        ])
-      )
+    async () => {
+      await delay(50);
+      return HttpResponse.json([
+        {
+          id: '9ef0aeb6-60bb-11ed-9b6a-0242ac120002',
+          title: 'january.pdf',
+        },
+        {
+          id: 'aed3c728-60bb-11ed-9b6a-0242ac120002',
+          title: 'february.pdf',
+        },
+        {
+          id: 'b60a4418-60bb-11ed-9b6a-0242ac120002 ',
+          title: 'march.pdf',
+        },
+      ]);
+    }
   ),
-  rest.get<null, GetDocumentsParams, GetDocumentsResponse>(
+  http.get<GetDocumentsParams, GetDocumentsResponse>(
     '/api/documents/:slug',
-    (request, response, context) =>
-      response(
-        context.delay(50),
-        context.status(200),
-        context.json(
-          new Array(3).fill(null).map((_, index) => ({
-            id: `${request.params.slug}-${index + 1}`,
-            title: `${request.params.slug}-document-${index + 1}.pdf`,
-          }))
-        )
-      )
+    async ({ params }) => {
+      await delay(50);
```

Comment:
@kettanaito would you recommend the usage of `delay(x)` in all/most handlers, or only occasionally?

I would think this is something you'd only want to use occasionally and. if so.. use the _realistic_ `delay()` ??

Reply:
Hard to recommend something, I'm not much aware of the intention here. 

Delaying the response must be a user's choice. I'd expect the user of this library to use `delay()` in whichever handlers they provide (if that's how it's architectured). 

You can use a realistic random delay as well. Can you tell me more about the intention here? Why is delay even needed?

Reply:
Nice, I didn't realize that we could call `delay()` without args to get a random amount of delay. This is awesome!